src/views/utility/views-common.js

```diff
@@ -26,3 +27,35 @@ export function removeDuplicates(originalArray, prop) {
   }
   return newArray;
 }
+
+export function getLinkForSegmentNumbers(language, segmentnr) {
+  let segmentlink = html`
+    ${segmentnr}
+  `;
+  if (language === 'pli') {
+    const cleanedSegment = segmentnr
+      .split(':')[1]
+      .replace(/_[0-9]+/g, '')
+      .replace('–', '--');
+    const linkText = segmentnr.match(/^tika|^anya|^atk/)
+      ? `https://www.tipitaka.org/romn/`
+      : `https://suttacentral.net/${
+          segmentnr.split(':')[0]
+        }/pli/ms#${cleanedSegment}`;
+    segmentlink = html`
+      <a target="_blanc" class="segment-link" href="${linkText}"
+        >${segmentnr}</a
+      >
+    `;
+  }
+  if (language === 'chn') {
+    const cleanedSegment = segmentnr.split(':')[0].replace(/_T/, 'n');
+    const linkText = `http://tripitaka.cbeta.org/${cleanedSegment}`;
+    segmentlink = html`
+      <a target="_blanc" class="segment-link" href="${linkText}"
+        >${segmentnr}</a
+      >
+    `;
+  }
+  return segmentlink;
+}
```

Comment:
I want to note that there is a slight conflict currently of two functions that do similar things: 
1. the newly introduced  getLinkForSegmentNumbers
2. replaceSegmentForDisplay of preprocessing.js
The second function is necessary in order to expand the filenames into more meaningful names (in the case of ACIP, the Kengyur filenames make hardly any sense for people who do not know ACIP well so we need to replace them with the Derge/Lhasa numbers). 
I suggest therefore to move the function getLinkForSegmentNumbers into preprocessing.js (in my eyes it makes a bit more sense since it is primarily a preprocessing step) and call replaceSegmentForDisplay inside getLinkForSegmentNumbers so the display is consistent on the website.

Reply:
There is no conflict as the getLinkForSegmentNumbers now only works for pali and chinese and not tibetan.
In the long run however it would be good if it could work for Tibetan as well and I noticed the conflict you are talking about in that case.

I'm just ta bit confused as to why replaceSegmentForDisplay is necessary. Should the backend not just give you back the right data to display? And taking things out of `window` seems very hacky to me as well.

As getLinkForSegmentNumbers is for all views and primarily for numbers and table, I think it should not be in the preprocessing.js but that is not a major point and I'm happy to move it if you think that's better.

Reply:
Regarding where to place it: I perceived preprocessing.js to be for preprocessing functions that are needed for all view-modes, but doesn't really matter. We can also move the replace-labels functions to views-common  and/or merge both files, preprocessing and views-common into one because they do the same anyway. 
I used window at that time because it's a global namespace and since we load the labels anyway when we load the menus, it was an easy way to get the labels quickly without need to redesign the entire code-structure. Of course we can do that on the backend, but then we need to put the labels into a separate value and transmit that together with the segments. The devil lies in the detail since it needs to be done on many places of the frontend independently, not just for labels but for filenames as well, and we will need for the Sanskrit at least and it's also done for the Chinese, so... But I am of course open to more elegant solutions.

Reply:
Okay, I see your point.
But are the labels not already in the backend for each file under "displayName"? It is simply a matter of retrieving that value also and showing it. I can take care of that.

Reply:
I think that should be it, yes. I think that retrieving these from the DB makes sense, the performance-gain by fetching the values from the menu is probably not worth the complexity in code. We just need to keep in mind that everywhere where either a segment or filename is shown in the UI, it actually get's properly replaced by the displayNames. and in my eyes that should be language-agnostic, since we do it in three languages now. i guess for Pali the displayNames are just identical with the fileNames/SegmentNrs?